src/sentry/api/invite_helper.py

```diff
@@ -170,19 +183,33 @@ def accept_invite(self, user=None):
         if self.member_already_exists:
             self.handle_member_already_exists()
             om.delete()
-        else:
-            om.set_user(user)
-            om.save()
-
-            create_audit_entry(
-                self.request,
-                actor=user,
-                organization=om.organization,
-                target_object=om.id,
-                target_user=user,
-                event=AuditLogEntryEvent.MEMBER_ACCEPT,
-                data=om.get_audit_log_data(),
-            )
+            return
+
+        try:
+            provider = AuthProvider.objects.get(organization=om.organization)
+        except AuthProvider.DoesNotExist:
+            provider = None
+
+        # AuthIdentity has a unique constraint on provider and user
+        auth_identities = AuthIdentity.objects.filter(auth_provider=provider, user=user)
```

Comment:
Since you never use the actual `AuthIdentity` rows here it's more efficient to use `.exists()`.